src/Umbraco.Web.BackOffice/Controllers/DictionaryController.cs

```diff
@@ -191,6 +193,40 @@ public ActionResult<DictionaryDisplay> GetById(Udi id)
             return _umbracoMapper.Map<IDictionaryItem, DictionaryDisplay>(dictionary);
         }
 
+        /// <summary>
+        /// Change the sort order for media
+        /// </summary>
+        /// <param name="move"></param>
+        /// <returns></returns>
+        public IActionResult PostMove(MoveOrCopy move)
+        {
+            var dictionaryItem =
+                _localizationService.GetDictionaryItemById(int.Parse(move.Id.ToString(), CultureInfo.InvariantCulture));
+
+            var parent =
+                _localizationService.GetDictionaryItemById(int.Parse(move.ParentId.ToString(), CultureInfo.InvariantCulture));
+
+            if (dictionaryItem == null)
+                return ValidationProblem("Dictionary item does not exist");
+
+            if (parent == null)
```

Comment:
Hi @Lantzify , OK I see your point, it's indeed trickier than it looks 😉 
I would go for your second check, because then we also cover the case where you might provide a `move.ParentId = 999999` which does not exist, in which case I think it is worth returning an error similar to what is done on line 209, instead of assuming a move under the root.

A few suggestions then: 
- there is something like an Umbraco "RootElementId" constant defined to -1 somewhere (don't remember exactly where, I can search if needed). So I would use that constant instead of hardcoded -1.
- I would keep the checks on dictionary item and parent separate, otherwize you will have to do the check on `parent == null` twice: here for the error and the just after it to define if you need to set `dictionaryItem.ParentId = null`

So, in the end, I would suggest something like this (pseudo-code 😁)

```
if (dictionaryItem == null) 
{
    Error "Dictionay Item does not exist" //should we provide the Id in the message?
}

if (parent == null)
{
    if (move.ParentId == UmbracoConstante.RootElementId)  // totally fake constant path
    {
        dictionaryItem.ParentId = null;
    }
    else
    {
        Error "Parent Item does not exist" //should we provide the Id in the message?
    }
}
else
{
    dictionaryItem.ParentId = parent.Key;
    // and here below we keep the existing stuff
}
    // and here below we keep the existing stuff

```
What do you think?

Reply:
Thank you! That looks really solid!